pkg/operator/controller/machineconfig/controller.go

```diff
@@ -0,0 +1,354 @@
+	// match with profiling MCP
+	ProfilingMCSelectorLabels = map[string]string{
+		"machineconfiguration.openshift.io/role": ProfilingMCPName,
+	}
+
+	// NodeSelectorLabels is for storing the labels to
+	// match the nodes to include in MCP
+	NodeSelectorLabels = map[string]string{
+		"node-role.kubernetes.io/worker": "",
+	}
+
+	// MachineConfigLabels is for storing the labels to
+	// add in machine config resources
+	MachineConfigLabels = map[string]string{
+		"machineconfiguration.openshift.io/role":                      ProfilingMCPName,
+		"machineconfigs.nodeobservability.olm.openshift.io/profiling": "",
+	}
+)
+
+//+kubebuilder:rbac:groups=nodeobservability.olm.openshift.io,resources=machineconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=nodeobservability.olm.openshift.io,resources=machineconfigs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=nodeobservability.olm.openshift.io,resources=machineconfigs/finalizers,verbs=update
+//+kubebuilder:rbac:groups=machineconfiguration.openshift.io,resources=machineconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=machineconfiguration.openshift.io,resources=kubeletconfigs,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=machineconfiguration.openshift.io,resources=machineconfigpools,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Machineconfig object Gagainst the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile
+func (r *MachineconfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.Log.Info("Reconciling MachineConfig of Nodeobservability operator")
+
+	// Fetch the nodeobservability.olm.openshift.io/machineconfig CR
+	r.CtrlConfig = &v1alpha1.Machineconfig{}
+	err := r.Get(ctx, req.NamespacedName, r.CtrlConfig)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			r.Log.Info("MachineConfig resource not found. Ignoring could have been deleted")
+			return ctrl.Result{RequeueAfter: defaultRequeueTime}, nil
+		}
+		// Error reading the object - requeue the request.
+		r.Log.Error(err, "failed to fetch MachineConfig")
+		return ctrl.Result{RequeueAfter: 3 * time.Minute}, err
+	}
+	r.Log.Info("MachineConfig resource found", "namespace", req.NamespacedName.Namespace, "name", req.NamespacedName.Name)
```

Comment:
should be in `context`

Reply:
comment incorporated in [d0805fd](https://github.com/openshift/node-observability-operator/pull/16/commits/d0805fd9c9a88344d99aa95ea52cf86ac8ad105c)